plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/mutate/MutateProcessorFactoryTests.java

```diff
@@ -50,32 +203,26 @@ public void testCreateGsubPattern() throws Exception {
         gsub.put("foo", Arrays.asList("\\s.*e\\s", "<word_ending_with_e>"));
         config.put("gsub", gsub);
 
-        Map<String, Tuple<Pattern, String>> compiledGsub = new HashMap<>();
-        Pattern searchPattern = Pattern.compile("\\s.*e\\s");
-        compiledGsub.put("foo", new Tuple<>(searchPattern, "<word_ending_with_e>"));
-
         MutateProcessor processor = factory.create(config);
-        for (Map.Entry<String, Tuple<Pattern, String>> entry : compiledGsub.entrySet()) {
-            Tuple<Pattern, String> actualSearchAndReplace = processor.getGsub().get(entry.getKey());
-            assertThat(actualSearchAndReplace, notNullValue());
-            assertThat(actualSearchAndReplace.v1().pattern(), equalTo(entry.getValue().v1().pattern()));
-            assertThat(actualSearchAndReplace.v2(), equalTo(entry.getValue().v2()));
-        }
+        assertThat(processor.getGsub().size(), equalTo(1));
+        GsubExpression gsubExpression = processor.getGsub().get(0);
+        assertThat(gsubExpression.getFieldName(), equalTo("foo"));
+        assertThat(gsubExpression.getPattern().pattern(), equalTo(Pattern.compile("\\s.*e\\s").pattern()));
```

Comment:
now that we have `GsubExpression` (hurray!). we should probably define an equals, where the `pattern` field is compared using the `.pattern()` command since that seems to be the proper way to check equivalence between `Pattern` objects.

then these assertions would be simplified as well.